lib/curl_gssapi.c

```diff
@@ -44,7 +44,26 @@ OM_uint32 Curl_gss_init_sec_context(
     const bool mutual_auth,
     OM_uint32 *ret_flags)
 {
-  OM_uint32 req_flags = GSS_C_REPLAY_FLAG;
+  gss_cred_id_t creds = GSS_C_NO_CREDENTIAL;
+  OM_uint32 major, discard, req_flags = GSS_C_REPLAY_FLAG;
+
+#ifdef HAVE_GSS_CRED_STORE
+  const char *ccache = data->set.str[STRING_KRB_CCACHE];
+
+  if(ccache != NULL) {
+    gss_key_value_element_desc store_elm = { "ccache", ccache };
+    gss_key_value_set_desc cred_store = { 1, &store_elm };
+
+    major = gss_acquire_cred_from(minor_status, GSS_C_NO_NAME,
+                                  GSS_C_INDEFINITE, GSS_C_NO_OID_SET,
+                                  GSS_C_INITIATE, &cred_store,
+                                  &creds, NULL, NULL);
+    if(GSS_ERROR(major)) {
+      infof(data, "Warning: gss_acquire_cred_from() failed\n");
```

Comment:
We'be better off having more information about this failure. What about using the gss call to turn major/minor into text?